darts/timeseries.py

```diff
@@ -21,7 +21,8 @@ class TimeSeries:
     def __init__(self,
                  df: pd.DataFrame,
                  freq: Optional[str] = None,
-                 fill_missing_dates: Optional[bool] = True):
+                 fill_missing_dates: Optional[bool] = True,
+                 columns: Optional[Union[List[str], str]] = None):
```

Comment:
I'm still not convinced that we need this argument to the init method. I think it's enough to simply re-use the column names of the dataframe, which would simplify everything. For instance the checks that column names are unique and of the correct size etc; Pandas can take care of that, and removing it for darts would improve separation of concerns IMO. It would still be good to keep this argument in `from_times_and_values()` method though, in order to build a DF with correct column names before building the TimeSeries.
WDYT?

Reply:
@grll what was teh final decision regarding this comment from @hrzn ?

Reply:
I agree that separation of concerns is good but we still need to enforce a few more things that is not guaranteed by pd.DataFrame such as uniqueness and str format. I created a private method in the latest commit to do that as it is used in a couple of places.